packages/node/src/policy-gen/policy-gen-compartment-class.js

```diff
@@ -163,48 +173,63 @@ const makeGetOverriddenResourceNames = (compartmentMap, resources) => {
         return
       }
 
-      /* c8 ignore next */
-      if (compartmentMap.entry.compartment === compartmentName) {
-        // This also "should never happen", as a foreign compartment should
-        // not be importing the entry compartment. Also, the entry
-        // compartment cannot have a canonical name, and we need a canonical
-        // name to apply policy
-        throw new ReferenceError(
-          `Unexpected entry compartment encountered in ${moduleDescriptorCompartment.label}; this is a bug`
-        )
-      }
-
       return moduleDescriptorCompartment
     }
+    const canonicalName = getCanonicalName(compartmentDescriptor)
```

Comment:
(old man yells at cloud) One appreciates good old hoisting after spending 10 minutes to figure out this is where the logic of getOverriddenResources begins.

no action required

Reply:
```suggestion

    const canonicalName = getCanonicalName(compartmentDescriptor)
```

Ok, maybe just this. It'd have helped.

Reply:
Now to the logic - There's no need to call getCanonicalName if we pass in the label from the same compartment where we're getting compartmentName.

Reply:
I've simplified this stuff per yr suggestions